sdk/identity/Azure.Identity/src/CredentialDiagnosticScope.cs

```diff
@@ -34,27 +36,26 @@ public AccessToken Succeeded(AccessToken token)
             return token;
         }
 
-        public AuthenticationFailedException FailAndWrap(Exception ex)
+        public Exception FailAndWrap(Exception ex)
```

Comment:
Your not actually ever returning from this method. Perhaps it should have a void return type and be renamed to FailAndThrow?

Reply:
The problem is that if `FailAndWrap` method doesn't return exception, compiler requires to return something from `catch` statement. So while method doesn't return anything, it is the easiest way for caller to tell that exception is thrown.